crypto/src/main/java/org/springframework/security/crypto/bcrypt/BCryptPasswordEncoder.java

```diff
@@ -90,11 +90,11 @@ public BCryptPasswordEncoder(BCryptVersion version, int strength) {
 	 * @param random   the secure random instance to use
 	 */
 	public BCryptPasswordEncoder(BCryptVersion version, int strength, SecureRandom random) {
-		if (strength != -1 && (strength < BCrypt.MIN_LOG_ROUNDS || strength > BCrypt.MAX_LOG_ROUNDS)) {
+		if (strength < BCrypt.MIN_LOG_ROUNDS || strength > BCrypt.MAX_LOG_ROUNDS) {
```

Comment:
Unfortunately, this breaks passivity for users who are providing the strength as -1. For that reason, unless we have a really good reason I don't think we can make this change.

Reply:
> Unfortunately, this breaks passivity for users who are providing the strength as -1

While I agree that this is true, I hardly could imaging a developer using -1 to get a default strength (Spring Security doesn't use it itself). At least, such behaviour is not documented. 
Javadoc comment for 'strength' parameter states that
`* @param strength the log rounds to use, between 4 and 31`

Moreover, to get a default strength a default constructor could be used BCryptPasswordEncoder(), which is natural, rather than BCryptPasswordEncoder(-1).

Also providing a default strength value directly in the statement
`this.strength = strength == -1 ? 10 : strength;`
skips the check
`(strength < BCrypt.MIN_LOG_ROUNDS || strength > BCrypt.MAX_LOG_ROUNDS)`

Reply:
Good points, @SvensenSeven, the developer would certainly have very reasonable workarounds.

However if allowing -1 isn't causing some kind of problem, then remaining passive needs to take priority.